source/crud/tests/unified/findOneAndReplace-dots_and_dollars.yml

```diff
@@ -0,0 +1,140 @@
+description: "findOneAndReplace-dots_and_dollars"
+
+schemaVersion: "1.0"
+
+createEntities:
+  - client:
+      id: &client0 client0
+      observeEvents: [ commandStartedEvent ]
+  - database:
+      id: &database0 database0
+      client: *client0
+      databaseName: &database0Name crud-tests
+  - collection:
+      id: &collection0 collection0
+      database: *database0
+      collectionName: &collection0Name coll0
+
+initialData: &initialData
+  - collectionName: *collection0Name
+    databaseName: *database0Name
+    documents:
+      - &initialDocument { _id: 1 }
+
+tests:
+  - description: "Replacing document with top-level dotted key on 3.6+ server"
+    runOnRequirements:
+      - minServerVersion: "3.6"
+    operations:
+      - name: findOneAndReplace
+        object: *collection0
+        arguments:
+          filter: { _id: 1 }
+          replacement: &dottedKey { _id: 1, a.b: 1 }
+        expectResult: *initialDocument
+    expectEvents: &expectEventsDottedKey
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                findAndModify: *collection0Name
+                query: { _id: 1 }
+                update: *dottedKey
+                new: { $$unsetOrMatches: false }
+    outcome:
+      - collectionName: *collection0Name
+        databaseName: *database0Name
+        documents:
+          - *dottedKey
+
+  - description: "Replacing document with top-level dotted key on pre-3.6 server yields server-side error"
+    runOnRequirements:
+      - maxServerVersion: "3.4.99"
+    operations:
+      - name: findOneAndReplace
+        object: *collection0
+        arguments:
+          filter: { _id: 1 }
+          replacement: *dottedKey
+        expectError:
+          isClientError: false
+    expectEvents: *expectEventsDottedKey
+    outcome: *initialData
+
+  - description: "Replacing document with dollar-prefixed key in embedded doc on 5.0+ server"
+    runOnRequirements:
+      - minServerVersion: "5.0"
+    operations:
+      - name: findOneAndReplace
+        object: *collection0
+        arguments:
+          filter: { _id: 1 }
+          replacement: &dollarPrefixedKeyInEmbedded { _id: 1, a: { $b: 1 } }
+        expectResult: *initialDocument
+    expectEvents: &expectEventsDollarPrefixedKeyInEmbedded
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                findAndModify: *collection0Name
+                query: { _id: 1 }
+                update: *dollarPrefixedKeyInEmbedded
+                new: { $$unsetOrMatches: false }
```

Comment:
These tests do not use `returnDocument`, which allows them to easily expect the initial document as a result.

According to the CRUD spec, the `new` option should be omitted if unspecified by the application (server defaults to false); however, I noticed that PHPLIB ran afoul of that (opened [PHPLIB-658](https://jira.mongodb.org/browse/PHPLIB-658) to track). I suppose other drivers might as well, so I'd like to be flexible here.

I reckon this is on par with what we're doing elsewhere in update tests for `upsert` and `multi` options (which have the same guidance in the CRUD spec).